src/adapter/src/flags.rs

```diff
@@ -26,7 +30,29 @@ pub fn compute_config(config: &SystemVars) -> ComputeParameters {
 }
 
 /// Return the current storage configuration, derived from the system configuration.
-pub fn storage_config(config: &SystemVars) -> StorageParameters {
+pub fn storage_config(
+    config: &SystemVars,
+    cluster_replica_sizes: &ClusterReplicaSizeMap,
+) -> StorageParameters {
+    // populating map of cluster size and corresponding memory limits in bytes where the
+    // value is given
+    let cluster_size_memory_map = cluster_replica_sizes
+        .0
+        .iter()
+        .filter_map(|(cluster_size, replica_allocation)| {
+            replica_allocation.memory_limit.map(|memory| {
+                (
+                    cluster_size.to_owned(),
+                    memory
+                        .0
+                        .as_u64()
+                        .try_into()
+                        .expect("should be able to cast to usize from u64"),
```

Comment:
```
use mz_ore::cast::CastFrom
```
let you do `u64::cast_from(<usize)`

Reply:
oh right, that's used in the rust <-> proto conversion as well.

Reply:
done